### PR TITLE
feat(source): support debezium avro format

### LINKED ISSUE #7892: Support DEBEZIUM AVRO format (with schema registry)

### PR DESCRIPTION
**Pull request by hzxa21**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Support debezium avro format with schema registery. Example:
```
CREATE table dbz (id integer primary key) with (
    connector = 'kafka',
    topic = 'dbserver1.inventory.customers',
    properties.bootstrap.server = '127.0.0.1:9092',
    scan.startup.mode = 'earliest'
) row format debezium_avro row schema location CONFLUENT SCHEMA REGISTRY 'http://localhost:8081';
```

Known limitations of this PR:

- We must use CREATE TABLE syntax with primary key specified to create dbz avro source
- Schema resolution fails when calling the from_avro_datum even when reader_schema == writer_schame. This is a limitation in apache-avro rust lib. We can ignore it for now since we haven't support source table schema evolution.

Close #7892 #7904 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
~~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).~~
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Connector (sources & sinks)

### Release note
```
CREATE TABLE [ IF NOT EXISTS ] source_name (
   pk_column_name data_type PRIMARY KEY
) 
WITH (
   connector='kafka',
   field_name='value', ...
) 
ROW FORMAT DEBEZIUM_AVRO
ROW SCHEMA LOCATION CONFLUENT SCHEMA REGISTRY '<schema_registry_url>';
```

### CODE REVIEW

File: src/connector/src/parser/mod.rs [hunk 1]
```diff
@@ -423,6 +430,17 @@ impl SpecificParserConfig {
             SourceFormat::Maxwell => SpecificParserConfig::Maxwell,
             SourceFormat::CanalJson => SpecificParserConfig::CanalJson,
             SourceFormat::Native => SpecificParserConfig::Native,
+            SourceFormat::DebeziumAvro => {
+                debug_assert!(info.use_schema_registry);
```
Comment (tabVersion) [inline src/connector/src/parser/mod.rs:434]: avro can be used without schema registry

**Review by tabVersion — APPROVED** [review]
basically LGTM
please add some tests

File: src/connector/src/parser/mod.rs (hunk 1 above)
Comment (hzxa21) [reply src/connector/src/parser/mod.rs:434]: debezium can be used without schema registry if JSON is used as the data format, which is the default option in debezium. Also, debezium remove its avro dependencies starting from version 2.0.0. However, mz doesn't even support JSON.

For AVRO data format, techinically it is possible to be used in debezium without schema registry if one provides the schema file and scarifies schema evolution but I doubt whether there is any real-world use cases. Therefore, I suggest we enforce schema registry for debezium avro and add back the schema file support only when needed in order no to confuse user.

**Comment by hzxa21** [comment]
> basically LGTM please add some tests

I have tested in my debezium enviroment and proved this PR working. I will add more unit tests in future PRs.

**Comment by hzxa21** [comment]
I have refined this PR a bit after testing with debezium avro enviroment. Both existing and newly inserted/updated rows in the source db is reflected in RW.
image
image


These are the known limitations of this PR:
1. We must use `CREATE TABLE` syntax with primary key specifies to create dbz avro source since we haven't supported upsert and primiary key derived from source.
2. The table fields should be specified in `CREATE TABLE`. @waruto210 will work on improving this part by extracting the table schema during `CREATE TABLE`.
3. Schema resolution fails when calling the `from_avro_datum` even when reader_schema == writer_schame. I think this is a limitation in `from_avro_datum` itself and I guess the reason is it cannot handle "Union" type schema, which is the case in dbz avro. I will try the `Reader` interface in apache-avro later but I think we can merge this PR first since we anyway won't handle schema evolution for now.

**Comment by waruto210** [comment]
> 3\. Schema resolution fails when calling the `from_avro_datum` even when reader_schema == writer_schame.

I also encountered this issue, even with the newest release version of `apache_avro`, but when can call `from_avro_datum` without `reader_schema`, though not very good, but it can parse out the Value.

**Comment by hzxa21** [comment]
> > 3. Schema resolution fails when calling the `from_avro_datum` even when reader_schema == writer_schame.
> 
> I also encountered this issue, even with the newest release version of `apache_avro`, but when can call `from_avro_datum` without `reader_schema`, though not very good, but it can parse out the Value.

Right. This is also what I did in this PR to workaround this issue.

**Pull request merged**